engine/src/test/scala/cromwell/engine/workflow/WorkflowActorSpec.scala

```diff
@@ -24,71 +23,106 @@ class WorkflowActorSpec extends CromwellTestkitSpec with WorkflowDescriptorBuild
     }
   })
 
-  private def createWorkflowActor(workflowId: WorkflowId = WorkflowId.randomId(),
-                                  startMode: StartMode = StartNewWorkflow,
-                                  wdlSource: WorkflowSourceFiles) = {
-    TestFSMRef(new WorkflowActor(workflowId, startMode, wdlSource, ConfigFactory.load, mockServiceRegistryActor))
-  }
+  val currentLifecycleActor = TestProbe()
+  val wdlSources = ThreeStep.asWorkflowSources()
+  val descriptor = createMaterializedEngineWorkflowDescriptor(WorkflowId.randomId(), workflowSources = wdlSources)
 
-  before {
-    val local = CromwellTestkitSpec.DefaultLocalBackendConfigEntry
-    CromwellBackends.initBackends(List(local), local, system)
+  private def createWorkflowActor(state: WorkflowActorState) = {
+    val actor = TestFSMRef(new WorkflowActor(WorkflowId.randomId(), StartNewWorkflow, wdlSources, ConfigFactory.load, mockServiceRegistryActor))
+    actor.setState(stateName = state, stateData = WorkflowActorData(Option(currentLifecycleActor.ref), Option(descriptor), StateCheckpoint(InitializingWorkflowState)))
+    actor
   }
 
-  "ShadowWorkflowActor" should {
-    "move from WorkflowUnstartedState to MaterializingWorkflowDescriptorState" in {
-      val wdlSources = WorkflowSourceFiles(HelloWorld.wdlSource("runtime { }"),
-        HelloWorld.rawInputs.toJson.toString(), "{ }")
-      val testActor = createWorkflowActor(wdlSource = wdlSources)
-      testActor.setState(stateName = WorkflowUnstartedState, stateData = WorkflowActorData.empty)
-      testActor ! StartWorkflowCommand
-      testActor.stateName should be (WorkflowActor.MaterializingWorkflowDescriptorState)
-      testActor.stop()
+  "WorkflowActor" should {
+
+    "run Finalization actor if Initialization fails" in {
+      val actor = createWorkflowActor(InitializingWorkflowState)
+
+      EventFilter.info(pattern = "transitioning from FinalizingWorkflowState to WorkflowFailedState", occurrences = 1).intercept {
+        within(CromwellTestkitSpec.timeoutDuration) {
+          actor ! WorkflowInitializationFailedResponse(Seq(new Exception("Materialization Failed")))
+        }
+      }
```

Comment:
I believe the `within` block should go outside the `EventFilter.info` block. Here, the `within` is only guaranteeing that the tell returns within the timeout.